### ISSUE TITLE
PowerPC DAGISel selects lwsync on unsupported targets.

### ISSUE THREAD

**Issue by winocm** [llvm/llvm-project#80896]
The PowerPC target selects the wrong instruction non-sequentially consistent atomic barriers are used on very old processor targets.

```
define i32 @ReadAcquire(ptr noundef %0) local_unnamed_addr {
  %2 = load atomic volatile i32, ptr %0 acquire, align 4
  ret i32 %2
}
```

This will get lowered into a lwsync when `-mcpu=603` is used with llc, but this instruction is not valid on either the 603 core, nor the 440 core. 

Refer to https://gcc.gnu.org/legacy-ml/gcc-patches/2012-07/msg01063.html for more information.

**Comment by chenzheng1030** [comment]
https://godbolt.org/z/41T1Yf9qo

Seems with latest llc, `lwsync` will not be generated, instead `isync` is used.

Would you please give your commands?

**Comment by winocm** [comment]
Yeah, it seems fine on trunk. The regression was fixed somewhere after v17.0.x.
